File: backend/spring-boot-petclinic/src/main/java/com/ntr/petclinic/entity/Pet.java
package com.ntr.petclinic.entity;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;
import java.util.Set;

@Entity
@Table(name = "pets")
@Getter
@Setter
public class Pet {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private Owner owner;

    @ManyToOne
    @JoinColumn(name = "type_id", nullable = false)
    private Type type;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "pet")
    private Set<Visit> visits;

    @Column(name = "name")
    private String name;

    @Column(name = "birth_date")
    private Date birthDate;

}

File: frontend/angular-petclinic/src/app/common/pet.ts
import { Owner } from './owner';
import { PetType } from './pettype';
import { Visit } from './visit';

export class Pet {
  [x: string]: any;
  id: string = '';
  name: string = '';
  birthDate: string = '';
  owner: Owner = new Owner();
  type: PetType = new PetType();
  visits: Visit[] = [];
}
